### PR TITLE
optimization for XCode 11 conversion script for XCode 11

### PR DESCRIPTION
**Pull request by sylvain-combe-sonarsource**
I implemented the tip suggested by @silverhammermba on #68 for conversion speed optimization on XCode 11.
It requires jq: https://stedolan.github.io/jq/ and I tested it successfully  Macos Big Sur and XCode SDK 11.3

### CODE REVIEW

**Comment by mcstoufer** [comment]
+1 on this for merge. Our conversion is taking hours instead of minutes.

File: swift-coverage/swift-coverage-example/xccov-to-sonarqube-generic.sh [hunk 1]
```diff
@@ -33,4 +38,49 @@ function xccov_to_generic {
...
+    exit -1
+  fi
+
+  # local reference=$(jq -r '.actions._values[2].actionResult.coverage.archiveRef.id._value'  tmp.json)
+  local reference=$(jq -r '.actions._values[]|[.actionResult.coverage.archiveRef.id],._values'  tmp.json | grep value | cut -d : -f 2 | cut -d \" -f 2)
+  if [ $? -ne 0 ]; then 
+    echo 'Failed to execute jq (https://stedolan.github.io/jq/)' 1>&2
+    exit -1
+  fi
+  xcrun xcresulttool export --type directory --path "$xccovarchive_file" --id "$reference" --output-path tmp.xccovarchive
```
Comment (mcstoufer) [inline swift-coverage/swift-coverage-example/xccov-to-sonarqube-generic.sh:72]: @sylvain-combe-sonarsource , if the $reference here ends up being a list of IDs (from a merged .xcresult bundle of multiple test plans), can we iterate over them and have the same output-path?

Comment (mcstoufer) [reply swift-coverage/swift-coverage-example/xccov-to-sonarqube-generic.sh:72]: Something like this shows promise:
```
for test_ref in $reference; do
     xcrun xcresulttool export --type directory --path "$xccovarchive_file" --id "$test_ref" --output-path tmp.xccovarchive
done
```

Comment (sylvain-combe-sonarsource) [reply swift-coverage/swift-coverage-example/xccov-to-sonarqube-generic.sh]: @mcstoufer
Many thanks for your review and this proposal. 
I was not able to test it fully as I don't have a project with multiple test plans or with merged .xcresult bundles.
But I've added your proposed change to the PR, please confirm it works for you and I'll merge it.

**Comment by sylvain-combe-sonarsource** [comment]
> +1 on this for merge. Our conversion is taking hours instead of minutes.

And I hope you meant "minutes instead of hours" :-)

**Comment by jarrodrobins** [comment]
+10000000000 on this PR. The existing conversion script is painfully slow.

**Comment by sylvain-combe-sonarsource** [comment]
> +10000000000 on this PR. The existing conversion script is painfully slow.

With this much weigh on your vote, I have no choice but to merge it :-)

**Pull request merged**